index.js

```diff
@@ -111,7 +111,23 @@ class Tonic extends window.HTMLElement {
     }
   }
 
-  _set (target, content = '') {
+  async _set (target, render, text = '') {
+    const name = render && render.constructor.name
+    let content = ''
+
+    if (name === 'AsyncFunction') {
+      content = await render.call(this) || ''
+    } else if (name === 'AsyncGeneratorFunction') {
+      for await (const value of render.call(this)) {
```

Comment:
`for await` only iterates `yield`ed values, any returned values are ignored.

For example, even though it looks totally valid, the following wouldn't work:

```js
  async * render () {
      yield this.html`Loading...`

      try { 
        const value = await this.getSomeData()
      } catch (e) {
        // early return
        return this.html `<div> Error: ${e.message} <div>`
      }
      // return final value:
      return this.html`
        <p>${value}</p>
      `
    }
```

But you can make it work with something like:

```js
while (true) {
  const { value, done } = await render.call(this)
  this._set(target, null, value)
  if (done) break
}
```

Reply:
introducing something like this could be cool but might be too magic. As a user my intuition would only expect `yield` to work in this case.

Reply:
interestingly I've had this convo about whether it's intuitive once already today: https://github.com/nodejs/node/pull/28501#discussion_r299988177 

this is an odd one because my intuition isn't matching others - to me, in every other scenario a return means supply a value - but in this one case with `for await` it means.. ignore my value. I kind of see `yield` like `stream.write` and `return` like `stream.end`.

Whatever we go with is ultimately fine, as long as it's documented

Reply:
hah! you make a really compelling argument with your code samples. I'm actually completely sold after reading those. Let's go with the while loop :)